tutorials/dnssec-cloud-dns-domains/index.md

```diff
@@ -0,0 +1,1074 @@
+domain registrar and the reseller.
+
+[4]: https://gwhois.org/
+[5]: https://gwhois.org/dns-example.info+dns
+
+~~~console
+$ whois dns-example.info
+Domain Name: DNS-EXAMPLE.INFO
+Registry Domain ID: D503300000000040442-LRMS
+Registrar WHOIS Server:
+Registrar URL: http://www.google.com
+…
+Registrar: Google Inc.
+Registrar IANA ID: 895
+…
+Reseller:
+…
+Name Server: NS-CLOUD-E1.GOOGLEDOMAINS.COM
+Name Server: NS-CLOUD-E2.GOOGLEDOMAINS.COM
+…
+~~~
+
+#### TLD *registries* vs. domain *registrars*
+
+Despite their similar names, TLD **registries** and domain **registrars** are
+separate organizations with different roles, although for .gov and .int TLDs,
+and some two-letter country code (ccTLD) top-level domains, the TLD registry may
+also provide domain registrar services.
+
+<dl>
+<dt>TLD registries</dt>
+<dd>
+    provide WHOIS and DNS name service for top-level domains (TLDs), and in most
+    cases, an API for domain registrars to manage delegation records and WHOIS
+    data.
+</dd>
+<dt>Domain registrars</dt>
+<dd>
+    provide retail services to customers purchasing or renewing domains, and
+    interact with TLD registries to manage delegation records in the TLD and
+    update WHOIS data for those domains. Many offer web pages for customers to
+    make those updates themselves, and some provide web APIs for automation
+    tools. Most also provide free DNS name service for domains that are
+    registered with them (although for domains delegated to Google Cloud DNS
+    name servers, those domain registrar name servers would not be used).
+</dd>
+<dt>Domain resellers</dt>
+<dd>
+    typically provide only the retail services of domain registrars, but may
+    provide web pages for updating delegation records.
+</dd>
+</dl>
+
+All three of these are shown on the right side of the overview diagram above; to
+avoid confusion, this tutorial uses the terms “*TLD* registries” and “*domain*
+registrars” throughout.
+
+## Costs
+
+This tutorial uses Google Cloud DNS, which has a monthly cost for the existence
+of each managed zone, as well as a per-query cost. Use the [Pricing
+Calculator][6] (scroll down on that page to see an estimate for Cloud DNS with
+one zone and up to several tens of thousands of incoming queries) to generate a
+cost estimate based on your projected usage.
+
+[6]: https://cloud.google.com/products/calculator/#id=cb34e9f9-3838-48c2-9346-3d20379aa285
+
+Although validating resolvers generate additional queries (for DNSKEY records)
+on DNSSEC-activated domains, the DNSKEY queries are negligible compared to
+normal traffic. If you enable DNSSEC with NSEC denial of existence (rather than
+the default NSEC3) recursive resolvers may be able to answer queries for
+non-existent domains without additional queries to your name servers, which
+might reduce the total number of queries (especially for denial-of-service
+attacks that send millions of queries for random non-existent subdomains).
+
```

Comment:
As I understand it, the purpose of the "Cost" section in these tutorials is not to do an exhaustive analysis of the factors that might influence the cost of operating a large cloud deployment, but rather to give the reader some idea of what it might actually cost them to try out the tutorial's examples.

It is probably already going a bit far to even mention the possibility of some potential benefit if you use NSEC denial of existence (the default for Google Cloud DNS when enabling DNSSEC is NSEC3). I think the simpler (and I hope now less awkward) description is sufficient for this tutorial.

I might try to write up another tutorial explaining in detail how to set up a multi-level zone structure with multiple DNSSEC signed zones: one using NSEC at the top level (containing all the names you don't care if anyone knows, like mail, www, etcetera) and others using NSEC3 below that where you can host trial features and other "unpublished" names where the fig leaf of NSEC3 might justify its use. In that tutorial, it would be appropriate to provide a detailed analysis such as you wrote (although probably not in the "Cost" section), so I will keep your suggestions in mind for that.